pkg/cli/image/mirror/mirror.go

```diff
@@ -183,6 +189,14 @@ func (o *MirrorImageOptions) Complete(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	// Regardless of what FilterByOS was passed or set, if KeepManifestList=true,
+	// have to set wildcard filter, otherwise the manifest list digest will be altered.
+	// This means that user will mirror all manifests in the list, as there is no way to
+	// preserve the manifest list digest while also modifying its constituents.
+	if o.KeepManifestList {
+		o.FilterOptions.FilterByOS = ".*"
```

Comment:
Maybe `klog.V(1)` here stating this fact.

Reply:
Alternatively, fail validation if these two conflict?

Reply:
yup, that was my first instinct - I changed to this because in discussions it was noted that it _is_ possible to have a valid combo of `--filter-by-os` with `--keep-manifest-list=true` , if and only if the filter value includes all manifest list os/arch options.  I will go back to the validation error, though, it's clearer. We can always update later if we find people complaining (pretty unlikely, i think).

Reply:
Updated with new Validation to error if/when `--keep-manifest-list=true` && `--filter-by-os=not/wildcard` with a clear error message.